### ISSUE TITLE
Add OS recognition/support for steps

### ISSUE THREAD

**Issue by faultyserver** [faultyserver/wrkflo#8]
Currently, `wrkflo` only _directly_ supports macOS. While this is fine in most cases, and most step definitions are portable enough to work on Linux, it would be nice to explicitly allow support for different OSes.

Some possible implementations:

```ruby
class SomeStep
  # alias, init, etc.

  def run
    # Using boolean flags is terse, but becomes verbose with multiple uses
    do_something      if windows?
    do_something_else if unix?

    # An `on` pattern is the easiest to read: "On Windows, do these things".
    on :windows do
      `C:\Program Files\Sublime Text 3\subl.exe #{config["path"]}`
      and_some_more
    end
    on :unix { `subl #{config["path"]}` }

    # Namespacing via a `with_os` block helps avoid collisions and follows the
    # old `respond_to` pattern from Rails, but quickly becomes verbose.
    with_os do |os|
      os.windows do
      end
      os.unix do
      end
    end

    # The simplest solution is to simply provide an OS constant.
    case OS do
    when :windows then do_something
    when :unix
      do_something_else
      and_some_more
    end
  end
end
```

My instinct is to implement the `on :some_os` pattern as well as the `os?` methods. The `on` pattern seems to be the most powerful without being overly verbose, and the `os?` methods can be used in combination with other conditionals for even better flow control.

**Comment by faultyserver** [comment]
The [`os` gem](https://github.com/rdp/os) seems to implement all of this in a pretty nice way.

I'm still a fan of adding the `on :os do` pattern on top of it, but it can easily depend on the gem for doing the actual OS detection.

**Comment by faultyserver** [comment]
Implemented by #10.
